src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new ArrayList[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        ensureCapacity();
+        int index = Math.abs(hash(key)) % buckets.length;
+        List<Entry<K, V>> bucket = buckets[index];
+        if (bucket == null) {
+            bucket = new ArrayList<>();
+            buckets[index] = bucket;
+        }
+        for (Entry<K, V> entry : bucket) {
+            if (key == null && entry.key == null
+                    || key != null && key.equals(entry.key)) {
+                entry.value = value;
+                return;
+            }
+        }
+        bucket.add(new Entry<>(key, value));
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = Math.abs(hash(key)) % buckets.length;
+        List<Entry<K, V>> bucket = buckets[index];
+        if (bucket != null) {
+            for (Entry<K, V> entry : bucket) {
+                if ((key == null && entry.key == null)
+                        || (key != null && key.equals(entry.key))) {
+                    return entry.value;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : key.hashCode() % buckets.length;
+    }
+
+    private void ensureCapacity() {
+        if ((double) size / buckets.length > LOAD_FACTOR) {
+            int newCapacity = buckets.length * GROW_FACTOR;
```

Comment:
let's reuse `put` here to simplify this method - save buckets to variable, set size to 0, clear buckets, iterate saved buckets and use put to save them again